catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBricksTest.java

```diff
@@ -43,98 +43,103 @@
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
 
 import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
 
-@RunWith(AndroidJUnit4.class)
+@Category({Cat.AppUi.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
 public class BroadcastBricksTest {
-	private String defaultMessage = "defaultMessage";
-	private int broadcastSendPosition = 1;
-	private int broadcastReceivePosition = 2;
-
 	@Rule
 	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
 			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
 
+	private static int whenBrickPosition = 0;
+	private static int broadcastSendPosition = 1;
+	private static int broadcastReceivePosition = 2;
+	private static int broadcastAndWaitPosition = 3;
+
+	@Parameters(name = "{2}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{broadcastSendPosition, R.id.brick_broadcast_spinner, "BroadcastSendTest"},
+				{broadcastReceivePosition, R.id.brick_broadcast_receive_spinner, "BroadcastReceiveTest"},
+				{broadcastAndWaitPosition, R.id.brick_broadcast_wait_spinner, "BroadcastAndWaitTest"}
+		});
+	}
+
+	@Parameter
+	public int brickPosition;
+
+	@Parameter(1)
+	public @IdRes int spinnerId;
+
+	@Parameter(2)
+	public String testName;
+
+	private String defaultMessage = "defaultMessage";
+
 	@Before
 	public void setUp() throws Exception {
-		createProjectAndGetStartScriptWithImages("BroadcastBrickTest");
+		createProject("BroadcastBricksTest");
 		baseActivityTestRule.launchActivity(null);
 	}
 
-	@Category({Cat.AppUi.class, Level.Smoke.class})
 	@Test
-	public void testBroadcastBricksLayout() {
-		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+	public void removeUnusedMessagesBroadcastTest() {
+		onBrickAtPosition(whenBrickPosition).checkShowsText(R.string.brick_when_started);
 		onBrickAtPosition(broadcastSendPosition).checkShowsText(R.string.brick_broadcast);
 		onBrickAtPosition(broadcastReceivePosition).checkShowsText(R.string.brick_broadcast_receive);
-	}
+		onBrickAtPosition(broadcastAndWaitPosition).checkShowsText(R.string.brick_broadcast_wait);
 
-	@Category({Cat.AppUi.class, Level.Functional.class})
-	@Test
-	@Flaky
-	public void testRemoveUnusedMessagesBroadcastSend() {
-		String uselessMessage = "useless";
-
-		createNewMessageOnSpinner(R.id.brick_broadcast_spinner, broadcastSendPosition, uselessMessage);
-
-		onBrickAtPosition(broadcastSendPosition).onSpinner(R.id.brick_broadcast_spinner)
-				.performSelect(defaultMessage);
+		onBrickAtPosition(brickPosition).onSpinner(spinnerId)
+				.checkShowsText(defaultMessage);
 
-		onView(withId(R.id.button_play))
+		String newMessage = "newMessage";
```

Comment:
please leave this in a separate method, it increases the readability by a lot, since this part here isnt actually the point of this test.